### DISCUSSION TITLE
Get Activity with Where Custom Properties Doesn't Work

### DISCUSSION THREAD

**Discussion by tandang213** [spatie/laravel-activitylog#728] [Q&A] [answered]
Hi, 

I've been log user's activity like login and logout like with this properties:

`withProperties([
                    'status' => 'login success',
                    'group' => 'user-activity',
                    'type' => 'auth',
                    'action' => 'login',
                    'user' => [
                        'email' => $email,
                        'id' => $id
                    ]
                ])
`

Then I am trying to get activity where type is 'auth' like this :
`
Activity::all()->where('properties->type','auth')
`
But it always return empty collection. 

Did I Do it wrong?

**Comment by Gummibeer** [comment, marked as answer]
Hey,
they issue is the order of your method calls. In your code-snippet you run the `where()` on the Eloquent Collection and not on the Eloquent Query-Builder.

```php
Activity::where('properties->type', 'auth')->get();
```
This will add a JSON field `WHERE` condition to the DB query and return a filtered Collection.
